### DISCUSSION TITLE
Feedback: How can we improve FlutterGPT!

### DISCUSSION THREAD

**Discussion by samyakkkk** [CommandDash/commanddash#182] [Q&A]
Dear community, 

We're building FlutterGPT with the vision of automating a majority of repetitive Flutter Development tasks. FlutterGPT is just getting started and we would love your feedback on how it could be improved. 

![giphy-3](https://github.com/Welltested-AI/fluttergpt/assets/56000318/7f746e1b-f1e4-475c-b923-b8c631310bb8)

For anything that comes to your mind (from annoying issues to possible improvements), please let us know!

**Comment by samyakkkk** [comment]
Populating some possible issues we can foresee (please upvote if this is the case for you):

**Comment by samyakkkk** [comment]
Refactor or optimization functions duplicate the entire file instead of replacing the selected code.

**Comment by samyakkkk** [comment]
The chat responses are not up to date and I had to google them instead.

**Comment by samyakkkk** [comment]
I'd like it if FlutterGPT can do code completion for me.

**Comment by faizan-jivani** [comment]
![Screenshot 2024-01-10 at 2 55 59 PM](https://github.com/Welltested-AI/fluttergpt/assets/45591301/89d952f4-62cc-4a9a-95ca-a81bc91f0ebf)
Hii @samyakkkk ,I just tried to generate the code from picture but I got this,
left side is image and right side is from auto generated code, it's not upto mark for now.

**Reply by samyakkkk** [reply]
@Zeeshan0201 seems like the image didn't upload successfully. could you please share again?

In general, we are waiting on Gemini Ultra models for the vision functions to get better.

**Reply by faizan-jivani** [reply]
[
![IMG_20240121_170602](https://github.com/Welltested-AI/fluttergpt/assets/45591301/5bd70a86-b145-4b7a-9338-d0512c633799)
](url)
Sure let me reupload

**Reply by samyakkkk** [reply]
Noted @Zeeshan0201. We'd brainstorm on how to further fix things here. A lot depends on Gemini Ultra access tbh since it is much more powerful.

**Reply by faizan-jivani** [reply]
Okay thanks 👍🏻
